app/src/main/java/unipd/se18/ocrcamera/ManualTestOnSinglePhoto.java

```diff
@@ -0,0 +1,212 @@
+             Start CameraActivity to take a new photo
+             */
+            Intent i = new Intent(ManualTestOnSinglePhoto.this, CameraActivity.class);
+            startActivity(i);
+        }
+    };
+
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        //UI initialization
+        setContentView(R.layout.activity_manual_test);
+        degreeEditText = findViewById(R.id.editText1);
+
+
+        //Get image path and text of the last image from preferences
+        SharedPreferences prefs = this.getSharedPreferences("prefs", MODE_PRIVATE);
+        String pathImage = prefs.getString("imagePath", null);
+        startingOCRText = prefs.getString("text", null);
+
+        final Bitmap photo = BitmapFactory.decodeFile(pathImage);
+
+        Button fab = findViewById(R.id.btnConfirm);
+        fab.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+
+                try {
+                    angleRotation = Integer.parseInt(degreeEditText.getText().toString());
+
+                    if (photo == null)
+                        throw new PhotoNullException();
+
+                    //Photo changes
+                    Bitmap photoToCompare = rotateImage(photo, angleRotation);
+
+                    // Instance of an OCR recognizer
+                    OCR ocrProcess = getTextRecognizer(TextRecognizer.Recognizer.mlKit,
+                            ocrListener);
+
+                    // Runs the operations of text extraction
+                    ocrProcess.getTextFromImg(photoToCompare);
+
+
+                } catch (PhotoNullException e) {
+                    Log.e(null, getResources().getString(R.string.errorLog1));
+                    Intent takeANewPhoto = new Intent(ManualTestOnSinglePhoto.this, CameraActivity.class);
+                    startActivity(takeANewPhoto);
+                }
+            }
+        });
+    }
+
+    /**Based on the text received and startingOCRTText, analise information like confidence
+     * and show them on textView
+     * @param text string got from OCR
+     * @modify degreeTextView, confidenceTextView, warningTextView and foundTextView
+     */
+    private void SetResult(String text){
+        String warning = findWarning(text, startingOCRText.length());
+        String confidence = Double.toString(compareStrings(startingOCRText,text));
+
+        degreeTextView.setText(angleRotation);
+        confidenceTextView.setText(confidence);
+        foundTextView.setText(text);
+        warningTextView.setText(warning);
+    }
+
+    /**
+     *Analise the given string and get the right warnings and add them on waring variable
+     * warning are found
+     * @param text String to analise
+     * @param length int of length of starting string
+     * @return  warnings
+     */
+    private String findWarning(java.lang.String text, int length) {
```

Comment:
Good to implement a warning but if I get more text from the rotated image I don't get any warning/message?
As is I get warning only if the new rotated image extract less characters than the original.